test/connection_test.exs

```diff
@@ -1,6 +1,36 @@
 defmodule Janus.ConnectionTest do
-  use ExUnit.Case
-  import Janus.HandlerTest.CallbackHelper
+  use ExUnit.Case, async: true
+  alias Janus.Connection
+  alias Janus.Connection.Transaction
+  alias Janus.Support.{ValidHandler, BrokenHandler, ValidTransport, BrokenTransport}
+
+  import Janus.Connection
+  import Janus.Support.Macro
+  import Mock
+  import Mox
+  import ExUnit.CaptureLog
+
+  @tag :tag
+  @receive_timeout 20_000
+  @receive_refute 2_000
+  @now %DateTime{
+    year: 2000,
+    month: 2,
+    day: 29,
+    zone_abbr: "CET",
+    hour: 23,
+    minute: 0,
+    second: 7,
+    utc_offset: 3600,
+    std_offset: 0,
+    time_zone: "Etc/UTC"
+  }
+
+  setup do
```

Comment:
A lot of tests here uses state. You can just make a state that fits most of those cases and put it into setup_all clause. It's executed once per file, and its result is found in any test.

Reply:
Done, but `Mox.stub/3` must be present in `setup` block. Also state that has `table` dependency must stay there.